pkg/common/cns-lib/volume/manager.go

```diff
@@ -1112,6 +1112,12 @@ func (m *defaultManager) deleteVolume(ctx context.Context, volumeID string, dele
 	volumeOperationRes := taskResult.GetCnsVolumeOperationResult()
 	if volumeOperationRes.Fault != nil {
 		faultType = ExtractFaultTypeFromVolumeResponseResult(ctx, volumeOperationRes)
+		// If volume is not found on host, but is present in CNS DB, we will get NotFound fault.
+		// Send back success as the volume is already deleted.
+		if IsNotFoundError(faultType) {
```

Comment:
we do not need to make changes in `deleteVolume`. Since improve idempotency is already released, we can just make the required change in `deleteVolumeWithImprovedIdempotency`